frontend/rust-lib/flowy-database2/src/services/field/type_options/timer_type_option/timer.rs

```diff
@@ -0,0 +1,171 @@
+use crate::entities::{TimerCellDataPB, TimerFilterPB};
+use crate::services::cell::{CellDataChangeset, CellDataDecoder};
+use crate::services::field::{
+  TimerCellData, TypeOption, TypeOptionCellDataCompare, TypeOptionCellDataFilter,
+  TypeOptionCellDataSerde, TypeOptionTransform,
+};
+use crate::services::sort::SortCondition;
+use collab_database::fields::{TypeOptionData, TypeOptionDataBuilder};
+use collab_database::rows::Cell;
+use flowy_error::FlowyResult;
+use serde::{Deserialize, Serialize};
+use std::cmp::Ordering;
+
+#[derive(Clone, Debug, Serialize, Deserialize, Default)]
+pub struct TimerTypeOption;
+
+impl TypeOption for TimerTypeOption {
+  type CellData = TimerCellData;
+  type CellChangeset = TimerCellChangeset;
+  type CellProtobufType = TimerCellDataPB;
+  type CellFilter = TimerFilterPB;
+}
+
+impl From<TypeOptionData> for TimerTypeOption {
+  fn from(_data: TypeOptionData) -> Self {
+    Self
+  }
+}
+
+impl From<TimerTypeOption> for TypeOptionData {
+  fn from(_data: TimerTypeOption) -> Self {
+    TypeOptionDataBuilder::new().build()
+  }
+}
+
+impl TypeOptionCellDataSerde for TimerTypeOption {
+  fn protobuf_encode(
+    &self,
+    cell_data: <Self as TypeOption>::CellData,
+  ) -> <Self as TypeOption>::CellProtobufType {
+    if let Some(minutes) = cell_data.0 {
+      return TimerCellDataPB {
+        minutes,
+        timer: TimerTypeOption::format_cell_data(minutes),
+      };
+    }
+    TimerCellDataPB {
+      minutes: i64::default(),
+      timer: "".to_string(),
+    }
+  }
+
+  fn parse_cell(&self, cell: &Cell) -> FlowyResult<<Self as TypeOption>::CellData> {
+    Ok(TimerCellData::from(cell))
+  }
+}
+
+impl TimerTypeOption {
+  pub fn new() -> Self {
+    Self
+  }
+
+  pub(crate) fn format_cell_data(minutes: i64) -> String {
```

Comment:
For this use-case, I'm wondering if we shouldn't do the parsing of the cell in the frontend itself.

Using Intl I believe we can do some pretty nice transformation, including support for different localizations.

See eg. https://github.com/andresaraujo/timeago.dart/blob/master/packages/timeago/lib/src/timeago.dart

Reply:
I wasn't sure about this either but doesn't this make it prone to inconsistency between different frontends?

Reply:
You're right, but I think in this case it's the better solution.

In the frontend we have locale information, as well as a better Intl library so we can much easilier parse the time into eg. `5y 100d 2h 31m`. With timestamps we also do parsing on the frontend for dates, we also are able to get timezone information on the frontend quite easily.

In Danish the equivalent would be `5år 100d 2t 31m`.